### ISSUE TITLE
SCANVI bug — classify_helper divides logits by their sum

### ISSUE THREAD

**Issue by marianogabitto** [scverse/scvi-tools#3804]
- Classifier.__init__ default = logits=False 
  - SCANVAE explicitly overrides the default to logits=True at _scanvae.py:142 
  - Therefore SCANVAE's self.classifier(z) returns raw logits 
  - Therefore the classify_helper factoring divides raw logits (which can be negative) by their sum (lines 779-791 in _base_module.py) 
            @auto_move_data
              def classify_helper(self, z):
                  if self.use_labels_groups:
                      w_g = self.classifier_groups(z)
                      unw_y = self.classifier(z)
                      w_y = torch.zeros_like(unw_y)
                      for i, group_index in enumerate(self.groups_index):
                          unw_y_g = unw_y[:, group_index]
                          w_y[:, group_index] = unw_y_g / (unw_y_g.sum(dim=-1, keepdim=True) + 1e-8)
                          w_y[:, group_index] *= w_g[:, [I]]
                  else:
                      w_y = self.classifier(z)
                  return w_y

```python
# Your code here
```

```pytb
[Paste the error output produced by the above code here]
```

#### Versions:

1.4.2

> VERSION

**Comment by ori-kron-wis** [comment]
@marianogabitto thanks. Looks like an old legacy bug (not because of creating the SupervisedModuleClass class) which only happens when use_labels_groups is True and logits is True also. I did not see such a case in our backtests so it was missed, I guess. Why did you need to use that?

I would assume the logits will be a parameter to the modules (scanvi/totalanvi/methylanvi) - but its hard coded inside, and I don't want to change it now (will perhaps impact existing pipelines ). They are True mainly for cross-entropy calculation.

Therefore, we removed the use_labels_groups logic in this PR. it looks redundant nowadays.
